src/app/modules/group/components/associated-activity/associated-activity.component.ts

```diff
@@ -73,10 +79,9 @@ export class AssociatedActivityComponent implements OnDestroy, ControlValueAcces
   }
 
   onRemove(): void {
-    if (this.rootActivityId === null) return;
+    if (this.rootActivity === null || this.state !== 'ready') return;
 
-    this.rootActivityId = null;
-    this.rootActivityName = null;
-    this.onChange(this.rootActivityId);
+    this.rootActivity = null;
+    this.onChange(null);
```

Comment:
on remove, aren't you supposed to publish "null" on your subject instead?

Reply:
and is it normal you call "onChange" only in onRemove?

Reply:
> on remove, aren't you supposed to publish "null" on your subject instead?

The subject is only here for the ui, only to fetch the activity name

> and is it normal you call "onChange" only in onRemove?

For now the root activity can only be removed from the group, so the only place where the onChange function is called is in the onRemove function

Reply:
> > on remove, aren't you supposed to publish "null" on your subject instead?
> 
> The subject is only here for the ui, only to fetch the activity name

But couldn't you have a race condition problem ? 
1. the item A is loaded (through activityChanged subject), loading done
1. you click on "remove"... it is very slow
1. the item B is loaded, (through activityChanged subject), loading done
2. The response of the remove (2) arrrives... the activity is cleared out

?

Reply:
I did consider this problem and thought because the only case that was handled for now was the remove (which doesn't do network calls, so it's pretty instantaneous) I could directly call the onChange function when the user clicks on the remove button.

I refactored the code, so the onChange is called in the subscribe of the subject now.

Reply:
why shouldn't the normal case trigger onChange?